src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs

```diff
@@ -270,16 +270,18 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
                 return NotFound();
             }
 
+            var menuContentAsJson = (JsonObject)menu.Content;
             // Look for the target menu item in the hierarchy.
-            var menuItem = FindMenuItem((JsonObject)menu.Content, menuItemId);
+            var menuItem = FindMenuItem(menuContentAsJson, menuItemId);
 
             // Couldn't find targeted menu item.
             if (menuItem == null)
             {
                 return NotFound();
             }
 
-            menu.Content.Remove(menuItemId);
```

Comment:
Do we have to change the code if `Remove` is added in this class?

Reply:
The problem here is not because of the Remove function, because `menu.Content` is a document object, so we can not directly remove a `MenuItem` from the document object based on the `menuId`
The code on the main branch here is wrong